### ISSUE TITLE
Index: Assigning dataset level permissions only partially indexess on save until new role added.

### ISSUE THREAD

**Issue by kcondon** [IQSS/dataverse#3039]
This affects builtin and shib users alike.
Scenario:
1. published dv
2. unpublished ds with 2 files
3. assign admin or curator roles to user to that ds
4. as that user, navigate to that dv and do not see ds card but can see files cards.
5. add a file downloader role to *any* user in that ds or dv, such as to dataverseAdmin
6. now see ds card with ds admin role.

I believe this happens with all role assignments, cards not appearing until another is added.
This happens with deleting/removing role assignments too.

This was happening in v4.3

**Comment by pdurbin** [comment]
Tentatively, the one line change below may be a fix (I'm on 0c13184) but I'm not sure if doing additional indexing will have a big performance impact. I'll study this some more:

```
murphy:dataverse pdurbin$ git diff src/main/java/edu/harvard/iq/dataverse/search/SolrIndexServiceBean.java
diff --git a/src/main/java/edu/harvard/iq/dataverse/search/SolrIndexServiceBean.java b/src/main/java/edu/harvard/iq/dataverse/search/SolrIndexServiceBean.java
index d8bf349..2c1a162 100644
--- a/src/main/java/edu/harvard/iq/dataverse/search/SolrIndexServiceBean.java
+++ b/src/main/java/edu/harvard/iq/dataverse/search/SolrIndexServiceBean.java
@@ -410,6 +410,7 @@ public class SolrIndexServiceBean {
         } else if (definitionPoint.isInstanceofDataset()) {
             // index the dataset itself
             indexPermissionsForOneDvObject(definitionPoint);
+            dvObjectsToReindexPermissionsFor.add(definitionPoint);
             // index files
             Dataset dataset = (Dataset) definitionPoint;
             for (DataFile datafile : filesToReIndexPermissionsFor(dataset)) {
murphy:dataverse pdurbin$ 
```

**Comment by pdurbin** [comment]
The way you can tell that the fix above helps is that the "Number of dvObject permissions indexed for" count changes from 0 to 1 if you run `/Applications/NetBeans/glassfish4/glassfish/bin/asadmin set-log-levels edu.harvard.iq.dataverse.search.IndexAsync=FINE`.

**Comment by pdurbin** [comment]
For now I put the fix above at https://github.com/pdurbin/dataverse/commit/fd566867f64539e04e354ab53f8dbc1441c7597f#commitcomment-16900492 since I was able to exercise this bug while working on #1012.

**Comment by pdurbin** [comment]
Fixed in pull request #3051. Passing to QA.

**Comment by kcondon** [comment]
This works, will check index all performance over night. Closing
